src/Tizen.System/PowerUsage/PowerUsage.cs

```diff
@@ -0,0 +1,224 @@
+            if (((PowerUsageError)ret != PowerUsageError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for resource" + (PowerUsageError)ret);
+                throw PowerUsageErrorFactory.ThrowPowerUsageException(ret);
+            }
+            return batteryUsage;
+        }
+
+        private static long ConvertDateTimeToTimestamp(DateTime dateTime)
+        {
+            DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+            return (long)((dateTime.ToUniversalTime() - epoch).TotalSeconds);
+        }
+
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <param name="rtypes">list of resource type identifiers like BLE, WiFi, CPU etc.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        /// <exception cref="ArgumentException">When an invalid parameter value is set.</exception>
+        /// <exception cref="UnauthorizedAccessException">If the privilege is not set.</exception>
+        /// <exception cref="InvalidOperationException">In case of any system error.</exception>
+        /// <exception cref="NotSupportedException">In case power usage is not supported</exception>
+        public static IList<double> GetPowerUsage(string appID, IList<ResourceType> rtypes, DateTime start, DateTime end)
+        {
+            IntPtr bmHandle = IntPtr.Zero;
+            IList<double> batteryUsage = new List<double>();
+            int ret = Interop.PowerUsage.GetPowerUsage(appID, ConvertDateTimeToTimestamp(start), ConvertDateTimeToTimestamp(end), bmHandle);
+            if (((PowerUsageError)ret != PowerUsageError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for all resources" + (PowerUsageError)ret);
+                throw PowerUsageErrorFactory.ThrowPowerUsageException(ret);
+            }
+
+            if (rtypes != null)
+            {
+                foreach (ResourceType type in rtypes)
+                {
+                    batteryUsage.Add(GetPowerUsage(bmHandle, type));
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Resource types parameter is empty");
+            }
+
+            ret = Interop.PowerUsage.Destroy(bmHandle);
```

Comment:
you should use `try/finally` to release `bmHandle` because some exceptions can be thrown before calling the Destroy() method.